beast/physicsmodel/priormodel.py

```diff
@@ -0,0 +1,240 @@
+import numpy as np
+from scipy.interpolate import interp1d
+from scipy.integrate import quad
+
+from beast.physicsmodel.grid_weights_stars import compute_bin_boundaries
+import beast.physicsmodel.priormodel_functions as pmfuncs
+
+
+__all__ = [
+    "PriorModel",
+    "PriorDustModel",
+    "PriorAgeModel",
+    "PriorMassModel",
+    "PriorMetallicityModel",
+    "PriorDistanceModel",
+]
+
+
+class PriorModel:
+    """
+    Compute the priors as weights given the input grid
+    """
+
+    def __init__(self, model, allowed_models=None):
+        """
+        Initialize with basic information
+
+        Parameters
+        ----------
+        model: dict
+          Choice of model type
+        """
+        if (allowed_models is not None) and (model["name"] not in allowed_models):
+            modname = model["name"]
+            raise NotImplementedError(f"{modname} is not an allowed model")
+        # save the model
+        self.model = model
+
+    def __call__(self, x):
+        """
+        Weights based on input model choice
+
+        Parameters
+        ----------
+        x : float
+            values for model evaluation
+        """
+        if self.model["name"] == "flat":
+            if "amp" in self.model.keys():
+                amp = self.model["amp"]
+            else:
+                amp = 1.0
+            if hasattr(x, 'shape'):
+                return np.full(x.shape, amp)
+            else:
+                return amp
+        elif self.model["name"] == "bins_histo":
+            # check if all ages within interpolation range
+            if np.all(
+                [np.max(x) <= cval <= np.min(x) for cval in self.model["values"]]
```

Comment:
values -> logages

Reply:
The change from `logages` -> `x` is needed to allow for most of the prior model code to be shared  between different parameters.  See the base PriorModel class at the top of priormodel.py.  Less code to maintain, with the price of a onetime change to the beast_settings file.